### ISSUE TITLE
Add Operator.__array__

### ISSUE THREAD

**Issue by adler-j** [odlgroup/odl#926]
While surfing through the documentation of [SPOT](http://www.cs.ubc.ca/labs/scl/spot/guide_quick2.html#7), I noticed that they use allow the syntax `double(op)`, which basically returns the matrix representation of `op`. I actually found that quite beautiful, and thought that we can do the same.

So the proposal is that we add `Operator.__array__` with the implementation 

```
def __array__(self):
    return matrix_representation(self)
 ```

what do the rest of you feel about this? I'm ambivalent myself, but I have to admit that syntax like

```
>>> op = wierd_big_operator(...)
>>> np.asarray(op)
array([[1, 2],
       [3, 4]])
```

would be quite neat. The biggest issue is ofc that it is *too* neat, and users may think it works even for more complicated cases (where it clearly will not be feasible).

**Comment by kohr-h** [comment]
I agree it's neat. My biggest issue with this version is that there is no way of documenting the fact that this is only for prototyping/debugging to the user. I'd prefer a normal method, perhaps `asarray` that shows its documentation to the user and warns about inefficiency.
